src/lib.rs

```diff
@@ -97,29 +143,80 @@ impl Office {
             Ok(Document { doc })
         }
     }
+    
+    /// Enable features such as password interaction
+    ///
+    /// # Arguments
+    ///  * `feature_flags` - The feature flags to set.
+    ///
+    /// @see [LibreOfficeKitOptionalFeatures]
+    ///
+    /// @since LibreOffice 6.0
+    pub fn set_optional_features(&mut self, optional_feature: LibreOfficeKitOptionalFeatures) -> Result<(), Error> {
+        unsafe {
+            (*self.lok_clz).setOptionalFeatures.unwrap()(self.lok, optional_feature as u64);
+            let error = self.get_error();
+            if error != "" {
+                return Err(Error::new(error));
+            }
+            Ok(())
+        }
+    }
+
     ///
     /// Set password required for loading or editing a document.
     ///
     /// Loading the document is blocked until the password is provided.
-    ///
+    /// This MUST be used in combination of features and within a callback
     ///
     /// # Arguments
     ///  * `url` - the URL of the document, as sent to the callback
     ///  * `password` - the password, nullptr indicates no password
     ///
-    /// In response to LOK_CALLBACK_DOCUMENT_PASSWORD, a valid password
+    /// In response to `LOK_CALLBACK_DOCUMENT_PASSWORD`, a valid password
     /// will continue loading the document, an invalid password will
-    /// result in another LOK_CALLBACK_DOCUMENT_PASSWORD request,
+    /// result in another `LOK_CALLBACK_DOCUMENT_PASSWORD` request,
     /// and a NULL password will abort loading the document.
     ///
-    /// In response to LOK_CALLBACK_DOCUMENT_PASSWORD_TO_MODIFY, a valid
+    /// In response to `LOK_CALLBACK_DOCUMENT_PASSWORD_TO_MODIFY`, a valid
     /// password will continue loading the document, an invalid password will
-    /// result in another LOK_CALLBACK_DOCUMENT_PASSWORD_TO_MODIFY request,
+    /// result in another `LOK_CALLBACK_DOCUMENT_PASSWORD_TO_MODIFY` request,
     /// and a NULL password will continue loading the document in read-only
     /// mode.
     ///
     /// @since LibreOffice 6.0
-
+    ///
+    /// # Example
+    ///
+    /// ``` 
+    /// use libreoffice_rs::{Office, LibreOfficeKitOptionalFeatures};
+    /// use std::error::Error;
+    /// use std::sync::atomic::{AtomicBool, Ordering};
+    /// use std::sync::Arc;
+    /// 
+    /// fn main() -> Result<(), Box<dyn Error>> {
+    ///     let input_path = "/home/vip/Tests/test.odt";
+    ///     let input_path_as_uri = "file:///home/vip/Tests/test.odt";
+    ///     let password = "test";
+    ///     let password_was_set = Arc::new(AtomicBool::new(false));
+    ///     let mut office = Office::new("/usr/lib/libreoffice/program")?;
+    /// 
+    ///     office.set_optional_features(LibreOfficeKitOptionalFeatures::LOK_FEATURE_DOCUMENT_PASSWORD)?;
+    ///     office.register_callback({
+    ///         let mut office = office.clone();
+    ///         move |_ntype, _payload| {
+    ///             if !password_was_set.load(Ordering::Relaxed) {
```

Comment:
I noticed that when incorrect password provided LibreOffice will not raise errors but invoke callback again, this `if` condition will causes process hangs, example like this kinda misleading.